### ISSUE TITLE
Bug: "text" attribute at labes / buttons can't be changed after returning from settings in kivy 2.1

### ISSUE THREAD

**Issue by xvasek** [kivy/kivy#7895]
**Software Versions**
* Python: 3.10.5
* OS: Windows 10 / Linux / android
* Kivy: 2.1 or master
* Kivy installation method: via pip

Hello, 

if you run the code below in kivy 2.1, when you press "Settings" button, and then "Close", the clock on the label "device_label" stops refreshing.

Expected behavior: The clock should refresh after closing settings.

Actually you can't change any text on existing labels / buttons on screen. It is possible to remove widgets and add new ones and then new ones bevahe correctly.

Interesting fact 1: You can change other properties on existing labels / buttons, ie. you can change font size. But text is "frozen".
Interesting fact 2: As you can see in the console, the "text" property of label is actually changing, but it is not visible in the application.

This bug appeared somewhere between kivy 2.0 (program behaves normally) and kivy 2.1 (bug exists) and is still present in current kivy master (v2.2.0.dev0)

```
# -*- coding: utf-8 -*-
import kivy
from kivy.app import App
from kivy.config import Config
from kivy.core.window import Window
from kivy.uix.label import Label
from kivy.uix.button import Button
from kivy.properties import ObjectProperty
from kivy.properties import StringProperty
from kivy.uix.boxlayout import BoxLayout
from kivy.storage.jsonstore import JsonStore
from kivy.base import EventLoop
from kivy.clock import Clock
from time import sleep
from datetime import datetime

class MyTopBar(BoxLayout):
    app = ObjectProperty(None)
    status = StringProperty("Ready", allownone = True)

    def __init__(self, **kwargs):
        def settings(button):
            self.app.open_settings()
        kwargs['orientation'] = "horizontal"
        kwargs['size_hint'] = (1, 0.1)
        super(MyTopBar, self).__init__(**kwargs)
        self.device_label = (Label(text = "Device: ", halign = "center"))
        self.status_label = (Label(text = self.status))
        self.button_settings = Button(text = "Settings")
        self.button_settings.bind(on_release = settings)
        self.add_widget(self.device_label)
        self.add_widget(self.status_label)
        self.add_widget(self.button_settings)

    def set_status(self, text):
        self.status = text
        self.status_label.text = self.status

    def show_clock(self):
        self.device_label.text = "Device: " + self.app.config.get('skate', 'device') + "\n" + datetime.now().strftime('%H:%M:%S')
        print (self.device_label.text)

class SKATE(App):
    
    def build_config(self, config):
        config.setdefaults('skate', {
            'device': '0',
            'font': 'small'
        })

    def build_settings(self, settings):
        jsondata = u"""
        [
            { "type": "numeric",
              "title": "Device",
              "desc": "Device number",
              "section": "skate",
              "key": "device" },

            { "type": "options",
              "title": "Font",
              "desc": "Velikost font size",
              "section": "skate",
              "key": "font",
              "options": ["small", "medium", "big", "huge"] }
              
        ]"""
        settings.add_json_panel('Skate',
            self.config, data=jsondata)

    def update_clock(self, delta):
        self.topbar.show_clock()

    def build(self):
        self.mainwindow = BoxLayout(orientation = "vertical")
        self.topbar = MyTopBar()
        self.topbar.app = self
        self.mainwindow.add_widget(self.topbar)
        Clock.schedule_interval(self.update_clock, 0.25)
        return self.mainwindow

if __name__ == '__main__':
    SKATE().run()
```

**Comment by gottadiveintopython** [comment]
With this complicated example, you can't be sure about the bug. If you try to simplify it, you'd discover the actual problem.

**Comment by misl6** [comment]
Should have been fixed via #8535 , please let us know if the issue persists on your side
